### ISSUE TITLE
Support for compiled JSLT template.

### ISSUE THREAD

**Issue by pratikpparikh** [schibsted/jslt#1]
Dear Author,

       Could you please look at the possibility of implementing a compiled templates?

Regards,
Pratik Parikh

**Comment by larsga** [comment]
The templates are compiled. When you call `Parser.compileString` what you get back is a template that's fully parsed and ready to be executed. It's even been through an optimizer. So in that sense it's compiled.

If you mean compiled to Java bytecode I have experimented with that, and found a 15-20% speedup. So the gains are not that large, at least not without additional tricks. It's not part of the current code, but it's something I intend to work on in the future.

**Comment by pratikpparikh** [comment]
@larsga yes I mean for compiled to Java bytecode and it would be great to have 15-20% speedup.

**Comment by larsga** [comment]
I will work on that, but it's a lot of work, and I think you'll find performance is pretty good already. I can run a large, complex transform at 150,000 transforms/second in a single thread on my laptop.

I will probably give priority to more work on the optimizer before I start working on bytecode compilation again, because there is some low-hanging fruit there. Having the optimizer handle those cases will benefit generated bytecode as well.

**Comment by pratikpparikh** [comment]
@larsga, I just thought about this as a feature (because XSLT and other transformation libraries have benefited from it in the past) but I agree other optimizations are more important.
